ensembl_genes/Red.py

```diff
@@ -53,30 +53,30 @@ def fetch_input(self):
         rpt = self.param_required('rpt')
         red_path = self.param_required('red_path')
         target_db_url = self.param_required('target_db_url')
-        self.param('target_db_url',target_db_url+'?local_infile=1')
+        self.param('target_db_url',f'{target_db_url}?local_infile=1')
 
         # make the temporary directory 'gnm' and copy the genome file into it
         # in this way we make sure that the only .fa file to be processed is the one we want
         try:
-            os.makedirs(gnm)
+            Path(gnm).mkdir()
         except PermissionError:
-            print('Could not create '+gnm+' directory.')
+            print(f'Could not create {gnm} directory.')
             raise
 
         # copy the genome file into the temporary directory
         # add suffix '.fa' to make sure it ends with '.fa' as required by Red
         # In this way we make sure that the only .fa file to be processed is the one we want
-        new_genome_file = gnm+os.path.sep+os.path.basename(genome_file)+'.fa'
+        new_genome_file = Path(f'{gnm}/{PurePath(genome_file).name}.fa')
         try:
             shutil.copyfile(genome_file,new_genome_file)
         except PermissionError:
-            print('Could not copy file '+genome_file+' into directory '+gnm)
+            print(f'Could not copy file {genome_file} into directory {gnm}')
             raise
 
         genome_file = self.param(genome_file,new_genome_file)
 
         # check that the Red binary exists
-        if not(os.path.isfile(red_path)):
+        if not(Path(red_path).exists()):
```

Comment:
I would create red_path as a Path object ahead of here and this would simply be:
```
if not red_path.exists():
```

Reply:
Done.